src/network/networkd-manager.c

```diff
@@ -1451,8 +1451,10 @@ void manager_free(Manager *m) {
         while ((pool = m->address_pools))
                 address_pool_free(pool);
 
-        set_free_with_destructor(m->rules, routing_policy_rule_free);
-        set_free_with_destructor(m->rules_foreign, routing_policy_rule_free);
+        /* routing_policy_rule_free() access m->rules and m->rules_foreign.
+         * So, it is necessary to set NULL after the sets are freed. */
+        m->rules = set_free_with_destructor(m->rules, routing_policy_rule_free);
+        m->rules_foreign = set_free_with_destructor(m->rules_foreign, routing_policy_rule_free);
```

Comment:
@yuwata thank you! I'm about to launch a VM to run the test there. In the meantime, I'm wondering why I couldn't reproduced the crash reliably. Judging by the code it should always crash. Looks like another mystery :-)